DSharpPlus.CommandsNext/CommandsNextExtension.cs

```diff
@@ -271,8 +271,7 @@ public Command FindCommand(string commandString, out string rawArguments)
                 if (!ignoreCase)
                     return null;
 
-                next = next.ToLowerInvariant();
-                var cmdKvp = this.RegisteredCommands.FirstOrDefault(x => x.Key.ToLowerInvariant() == next);
+                var cmdKvp = this.RegisteredCommands.FirstOrDefault(x => x.Key.Equals(next, StringComparison.InvariantCultureIgnoreCase));
```

Comment:
Since there are a lot of changes, i guess it will be easier to post screenshots with repro steps on each individual change one by one.
For this one you just need to run `sudo` command in test bot, with command name written in different cases, since this check runs when trying to find command by its name, in different case than its stored in RegisteredCommands dictionary when ignoring case is enabled.
`d# sudo @testingbot AvAtar`  
![image](https://user-images.githubusercontent.com/50947930/152695839-59735c4e-9ad3-4733-94a9-970895c673b5.png)